File: backend/mcp_server/tools/cases/logs.py
"""案件进展日志 MCP tools"""

from __future__ import annotations

from typing import Any

from mcp_server.client import client


def list_case_logs(case_id: int) -> list[dict[str, Any]]:
    """查询指定案件的所有进展日志，按时间倒序排列。"""
    return client.get("/cases/logs", params={"case_id": case_id})  # type: ignore[return-value]


def create_case_log(case_id: int, content: str) -> dict[str, Any]:
    """为案件添加进展日志。content 为日志内容，支持多行文本。"""
    return client.post("/cases/logs", json={"case_id": case_id, "content": content})  # type: ignore[return-value]

File: frontend/src/features/cases/api/logs.ts
import { createFeatureApiClient } from '@/lib/api'
import type { CaseLog, CaseLogAttachment, CaseNumber } from '../types'

const client = createFeatureApiClient('cases')

export const logsApi = {
  list: async (caseId: number | string): Promise<CaseLog[]> =>
    client.get('logs', { searchParams: { case_id: String(caseId) } }).json<CaseLog[]>(),

  listAll: async (): Promise<CaseLog[]> =>
    client.get('logs').json<CaseLog[]>(),

  create: async (data: { case_id: number; content: string; reminder_type?: string; reminder_time?: string }): Promise<CaseLog> =>
    client.post('logs', { json: data }).json<CaseLog>(),

  update: async (id: number | string, data: { case_id?: number; content?: string }): Promise<CaseLog> =>
    client.put(`logs/${id}`, { json: data }).json<CaseLog>(),

  delete: async (id: number | string): Promise<void> => {
    await client.delete(`logs/${id}`)
  },

  uploadAttachments: async (logId: number | string, files: File[]): Promise<CaseLogAttachment[]> => {
    const formData = new FormData()
    files.forEach((file) => formData.append('files', file))
    return client.post(`logs/${logId}/attachments`, { body: formData }).json<CaseLogAttachment[]>()
  },

  listCaseNumbers: async (caseId: number | string): Promise<CaseNumber[]> =>
    client.get('case-numbers', { searchParams: { case_id: String(caseId) } }).json<CaseNumber[]>(),

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  createCaseNumber: async (data: any): Promise<CaseNumber> =>
    client.post('case-numbers', { json: data }).json<CaseNumber>(),

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  updateCaseNumber: async (id: number | string, data: any): Promise<CaseNumber> =>
    client.put(`case-numbers/${id}`, { json: data }).json<CaseNumber>(),

  deleteCaseNumber: async (id: number | string): Promise<void> => {
    await client.delete(`case-numbers/${id}`)
  },
}
